adapters/InMobi/InMobiAdapter/GADMAdapterInMobiBannerAd.m

```diff
@@ -0,0 +1,194 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#import "GADMAdapterInMobiBannerAd.h"
+#import <InMobiSDK/IMSdk.h>
+#include <stdatomic.h>
+#import "GADInMobiExtras.h"
+#import "GADMAdapterInMobiConstants.h"
+#import "GADMAdapterInMobiInitializer.h"
+#import "GADMAdapterInMobiUtils.h"
+#import "GADMInMobiConsent.h"
+#import "GADMediationAdapterInMobi.h"
+
+static CGSize GADMAdapterInMobiSupportedAdSizeFromGADAdSize(GADAdSize gadAdSize) {
+    NSArray<NSValue *> *potentialSizeValues =
+    @[ @(GADAdSizeBanner), @(GADAdSizeMediumRectangle), @(GADAdSizeLeaderboard) ];
+    
+    GADAdSize closestSize = GADClosestValidSizeForAdSizes(gadAdSize, potentialSizeValues);
+    return CGSizeFromGADAdSize(closestSize);
+}
+
+@implementation GADMAdapterInMobiBannerAd {
+    
+    id<GADMediationBannerAdEventDelegate> _bannerAdEventDelegate;
+    
+    /// Ad Configuration for the banner ad to be rendered.
+    GADMediationBannerAdConfiguration *_bannerAdConfig;
+    
+    GADMediationBannerLoadCompletionHandler _bannerRenderCompletionHandler;
+    
+    /// InMobi banner ad object.
+    IMBanner *_adView;
+    
+}
+
+- (void)loadBannerForAdConfiguration:(nonnull GADMediationBannerAdConfiguration *)adConfiguration completionHandler:(nonnull GADMediationBannerLoadCompletionHandler)completionHandler {
+    _bannerAdConfig = adConfiguration;
+    __block atomic_flag completionHandlerCalled = ATOMIC_FLAG_INIT;
+    
+    __block GADMediationBannerLoadCompletionHandler originalCompletionHandler =
+    [completionHandler copy];
+    _bannerRenderCompletionHandler = ^id<GADMediationBannerAdEventDelegate>(_Nullable id<GADMediationBannerAd> bannerAd, NSError *_Nullable error) {
+        if (atomic_flag_test_and_set(&completionHandlerCalled)) {
+            return nil;
+        }
+        id<GADMediationBannerAdEventDelegate> delegate = nil;
+        if (originalCompletionHandler) {
+            delegate = originalCompletionHandler(bannerAd, error);
+        }
+        originalCompletionHandler = nil;
+        return delegate;
+    };
+    
+    
+    GADMAdapterInMobiBannerAd *__weak weakSelf = self;
+    NSString *accountID = _bannerAdConfig.credentials.settings[GADMAdapterInMobiAccountID];
+    [GADMAdapterInMobiInitializer.sharedInstance
+     initializeWithAccountID:accountID
+     completionHandler:^(NSError *_Nullable error) {
+        GADMAdapterInMobiBannerAd *strongSelf = weakSelf;
+        if (!strongSelf) {
+            return;
+        }
+        
+        if (error) {
+            NSLog(@"[InMobi] Initialization failed: %@", error.localizedDescription);
+            strongSelf->_bannerRenderCompletionHandler(nil, error);
+            return;
+        }
+        [strongSelf requestBannerWithSize:strongSelf->_bannerAdConfig.adSize];
+    }];
+}
+
+- (void)requestBannerWithSize:(GADAdSize)adSize {
```

Comment:
Nit, prefer renaming `adSize` to `requestedAdSize` to avoid confusion.

Reply:
done